bubble_sort.rb

```diff
@@ -0,0 +1,23 @@
+def bubble_sort(arr)
+  for i in 0...arr.length
+    for j in 0...arr.length - i - 1
+      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
+    end
+  end
+  arr
+end
+
+def bubble_sort_by(arr)
+  for i in 0...arr.length
+    for j in 0...arr.length - i - 1
```

Comment:
### Optional change

Ruby is an amazing language (I think I like it more than JS) And they have a "ruby way" of doing things. One of them is not liking the use of for loops, because they have some amazing methods instead!

In this case and the other for loop you have, you could use:

```
array.each_with_index do |e, i|
     for j in 0...arr.length - i - 1
      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
    end
end 
```

Or try to play with a while loop instead.